src/EnergyPlus/UnitarySystem.cc

```diff
@@ -7067,13 +7102,154 @@ namespace UnitarySystems {
+                std::string loc_cooling_coil_object_type("COIL:COOLING:WATER:DETAILEDGEOMETRY");
+                if (UtilityRoutines::SameString(loc_cooling_coil_object_type, input_specs.cooling_coil_object_type)) {
+                    input_specs.cooling_supply_air_flow_rate_method = UtilityRoutines::MakeUPPERCase("SupplyAirFlowRate");
+                    input_specs.cooling_supply_air_flow_rate = DataSizing::AutoSize;
+                }
+                // optional input fields
+                Real64 minAir2FluidTempOffset(0.0);
+                if (fields.find("minimum_air_to_water_temperature_offset") != fields.end()) { // not required field, has default value of 0.0
+                    minAir2FluidTempOffset = fields.at("minimum_air_to_water_temperature_offset");
+                }
+                std::string dehumidControlType("None");
+                if (fields.find("dehumidification_control_type") != fields.end()) {
+                    dehumidControlType = UtilityRoutines::MakeUPPERCase(AsString(fields.at("dehumidification_control_type")));
+                }
+                input_specs.dehumidification_control_type = dehumidControlType;
+
+                std::string runOnSensibleLoad("YES");
+                if (fields.find("run_on_sensible_load") != fields.end()) {
+                    runOnSensibleLoad = UtilityRoutines::MakeUPPERCase(AsString(fields.at("run_on_sensible_load")));
+                }
+                std::string runOnLatentLoad("NO");
+                if (fields.find("run_on_latent_load") != fields.end()) {
+                    runOnLatentLoad = UtilityRoutines::MakeUPPERCase(AsString(fields.at("run_on_latent_load")));
+                }
+
+                if (runOnSensibleLoad == "YES" && runOnLatentLoad == "NO") {
+                    input_specs.latent_load_control = "SensibleOnlyLoadControl";
+                } else if (runOnSensibleLoad == "NO" && runOnLatentLoad == "YES") {
+                    input_specs.latent_load_control = "LatentOnlyLoadControl";
+                } else if (runOnSensibleLoad == "YES" && runOnLatentLoad == "YES") {
+                    // not sure if this is control type is allowed
+                    input_specs.latent_load_control = "LatentOrSensibleLoadControl";
+                }
+
+                // now translate to UnitarySystem
+                thisSys.UnitType = cCurrentModuleObject;
+                input_specs.control_type = "Setpoint";
+                thisSys.m_CoolCoilExists = true; // is always true
+                thisSys.m_LastMode = state.dataUnitarySystems->CoolingMode;
+                // set water-side economizer temperature offset
+                thisSys.m_minAirToWaterTempOffset = minAir2FluidTempOffset;
+                // set water-side economizer flag
+                thisSys.m_waterSideEconomizerFlag = true;
+
+                thisSys.processInputSpec(state, input_specs, sysNum, errorsFound, ZoneEquipment, ZoneOAUnitNum);
```

Comment:
New CoilSystem:Cooling:Water inputs are processed normally in processInputsSpec and then inputs specific to the heat recovery model are processed after this point.